back/src/controller/router.ts

```diff
@@ -1,74 +1,96 @@
-import { Router } from 'express';
-import { BusinessLogic } from '../interfaces/business';
+import { Router } from "express";
+import { BusinessLogic } from "../interfaces/business";
+import { TransferRequest } from "../entities/TransferRequest";
 
 // force BigInt to be serialized as string in the JSON serialization
-(BigInt.prototype as any).toJSON = function() { return this.toString() }
+(BigInt.prototype as any).toJSON = function () {
+  return this.toString();
+};
 
 const createRouter = (bl: BusinessLogic) => {
   // create an express router and export it
   const router = Router();
 
   // default route returns a ok status
-  router.get('/', (req, res) => {
+  router.get("/", (req, res) => {
     res.sendStatus(200);
   });
 
-
-  router.get('/transactions/:id', async (req, res) => {
+  router.get("/transactions/:id", async (req, res) => {
     console.log("Calling getTransaction", req.params.id);
     try {
       const t = await bl.web3.getTransaction(req.params.id);
-      console.log("Transaction found", t)
+      console.log("Transaction found", t);
       res.json(t);
     } catch (error: unknown) {
-      res.status(500).json({error: (error as Error).message});
+      res.status(500).json({ error: (error as Error).message });
       console.error("Error in getTransaction", error);
     }
   });
 
-  router.post('/transfer', async (req, res) => {
+  router.post("/transfer", async (req, res) => {
     try {
       // TODO: implement a crypto transfer here
-      const hash = "0x1234567890123456789012345678901234567890123456789012345678901234"
-      res.json({hash})
+      const request: TransferRequest = req.body;
```

Comment:
Initializing a class that has a constructor with a simple assignment seems wrong. Also, the req.body will not hold a bigint so conversion is needed